lib/dynamo-restore.js

```diff
@@ -89,22 +90,23 @@ DynamoRestore.prototype._validateS3Backup = function(options) {
 };
 
 DynamoRestore.prototype._validateTable = function(options) {
-    var dynamodb = new AWS.DynamoDB();
+    var dynamodb = this.dynamodb;
     if (!options.table) {
         return this.emit('error', 'Please provide a Dynamo DB table name to restore to.');
     }
     dynamodb.listTables({}, this._checkTableExists.bind(this));
 };
 
 DynamoRestore.prototype._checkTableExists = function(error, data) {
+    var dynamodb = this.dynamodb;
     if (error || !data || !data.TableNames) {
         return this.emit('error', 'Fatal Error. Could not connect to AWS DynamoDB engine. Please check your credentials.');
     }
     if (data.TableNames.indexOf(this.options.table) > -1) {
         // Table exists, should we overwrite it??
         if (this.options.overwrite) {
-            this.emit('warning', util.format('WARN: table [%s] will be overwritten.', options.table));
-            dynamodb.describeTable({ TableName: this.options.table }, this._checkTableReady.bind(this));
+            this.emit('warning', util.format('WARN: table [%s] will be overwritten.', this.options.table));
+            setTimeout(dynamodb.describeTable.bind(dynamodb, { TableName: this.options.table }, this._checkTableReady.bind(this)), 1000);
```

Comment:
Hiya, this defers writing to table to allow S3 stream interface (this.readline) to be defined.